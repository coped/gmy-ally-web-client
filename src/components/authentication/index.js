import Login from "components/authentication/Login";
import LoginForm from "components/authentication/LoginForm";
import SignupForm from "components/authentication/SignupForm";

export { Login, LoginForm, SignupForm };
