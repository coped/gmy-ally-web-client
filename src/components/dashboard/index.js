import Dashboard from "components/dashboard/Dashboard";

export { Dashboard };
