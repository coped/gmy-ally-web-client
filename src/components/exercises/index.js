import Exercises from "components/exercises/Exercises";

export { Exercises };
