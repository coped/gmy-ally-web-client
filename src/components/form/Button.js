import React from "react";

export default function Button({
  children,
  classList = [],
  loading,
  onClick = undefined,
}) {
  const classes = () => {
    const list = ["button", ...classList];
    return loading ? [...list, "is-loading"].join(" ") : list.join(" ");
  };
  return (
    <div className="field">
      <div className="control">
        <button className={classes()} onClick={onClick}>
          {children}
        </button>
      </div>
    </div>
  );
}
