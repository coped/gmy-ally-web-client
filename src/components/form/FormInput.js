import React from "react";

export default function Input({ label, ...rest }) {
  return (
    <div className="field">
      <label className="label">{label}</label>
      <div className="control">
        <input className="input" {...rest} />
      </div>
    </div>
  );
}
