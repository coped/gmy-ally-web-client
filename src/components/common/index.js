import Notification from "components/common/Notification";
import Button from "components/common/Button";
import PageLoader from "components/common/PageLoader";

export { Notification, Button, PageLoader };
