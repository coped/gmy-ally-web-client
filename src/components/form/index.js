import Button from "components/form/Button";
import Input from "components/form/Input";

export { Button, Input };
