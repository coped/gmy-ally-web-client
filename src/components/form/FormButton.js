import React from "react";
import { Button } from "components/common";

export default function formButton({ children, ...rest }) {
  return (
    <div className="field">
      <div className="control">
        <Button {...rest}>{children}</Button>
      </div>
    </div>
  );
}
