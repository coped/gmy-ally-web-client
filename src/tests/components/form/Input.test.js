import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Input } from "components/form";

Enzyme.configure({ adapter: new Adapter() });

describe("Input", () => {
  const props = {
    label: "Label:",
    name: "inputName",
    text: "text",
    value: "Input value.",
    onChange: () => true,
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Input />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders with props without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Input {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("correctly renders props", () => {
    const wrapper = shallow(<Input {...props} />);
    expect(wrapper.find("label").text()).toEqual(props.label);
    expect(wrapper.find("input").props().value).toEqual(props.value);
  });

  it("properly calls onChange function on input change", () => {
    const mockCallback = jest.fn();
    const theseProps = { ...props, onChange: mockCallback };
    const wrapper = shallow(<Input {...theseProps} />);
    wrapper.find("input").simulate("change");
    expect(mockCallback.mock.calls.length).toEqual(1);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<Input {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
