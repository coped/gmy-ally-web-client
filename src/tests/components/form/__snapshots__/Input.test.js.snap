// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input has a valid snapshot 1`] = `
<div
  className="field"
>
  <label
    className="label"
  >
    Label:
  </label>
  <div
    className="control"
  >
    <input
      className="input"
      name="inputName"
      onChange={[Function]}
      type="text"
      value="Input value."
    />
  </div>
</div>
`;
